src/Workspaces/Core/Portable/SymbolId/SymbolKey.BodyLevelSymbolKey.cs

```diff
@@ -63,6 +63,10 @@ public override SymbolKeyResolution Resolve(Compilation compilation, bool ignore
             private static IEnumerable<ValueTuple<TSymbol, int>> EnumerateSymbols(Compilation compilation, ISymbol containingSymbol, string name, CancellationToken cancellationToken)
             {
                 int ordinal = 0;
+                if (compilation == null)
```

Comment:
[GetSymbolKey](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/SymbolId/SymbolKeyExtensions.cs#L11) has been doing this forever.  I don't know why we ever thought compilation would *not* be null.